compiler/rustc_target/src/asm/csky.rs

```diff
@@ -99,10 +101,6 @@ def_regs! {
         f29: freg = ["fr29","vr29"],
         f30: freg = ["fr30","vr30"],
         f31: freg = ["fr31","vr31"],
-        #error = ["r7", "l3"] =>
-            "the base pointer cannot be used as an operand for inline asm",
-        #error = ["r8","l4"] =>
-            "the frame pointer cannot be used as an operand for inline asm",
```

Comment:
Are you sure this is correct? https://github.com/llvm/llvm-project/blob/9322216021f1d8241b87a81c12321b05aa6d1cfb/llvm/lib/Target/CSKY/CSKYRegisterInfo.cpp#L50-L56 still lists them as reserved in case of variable sized objects on the stack.

Reply:
yeah, I'm sure. `r7 ` and `r8` is used for abi v1 but not  abi v2. I have checked the document of csky and confirmed it.

Reply:
It doesn't matter whether or not it is used by the abi across functions. What matters is if LLVM reserves it for internal use within the function or not. There is nothing in the x86_64 abi about rbx being reserved either, but LLVM reserves the right to use it in certain cases anyway so we have to deny usage of it in inline asm.

Reply:
Ok, you are right. Thanks for that!